src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java

```diff
@@ -0,0 +1,393 @@
+                if (building == null)
+                {
+                    Log.getLogger().error(
+                      String.format("Worker (%d:%d) ERROR - Starting and missing building(%s)",
+                        worker.getColony().getID(), worker.getCitizenData().getId(), wo.getBuildingLocation()));
+                    return;
+                }
+
+                worker.sendLocalizedChat(COM_MINECOLONIES_COREMOD_ENTITY_BUILDER_BUILDSTART, job.getStructure().getName());
+
+                //Don't go through the CLEAR stage for repairs and upgrades
+                if (building.getBuildingLevel() > 0)
+                {
+                    wo.setCleared(true);
+                }
+            }
+            else if(!(wo instanceof WorkOrderBuildMiner))
+            {
+                worker.sendLocalizedChat(COM_MINECOLONIES_COREMOD_ENTITY_BUILDER_BUILDSTART, wo.getName());
+            }
+        }
+    }
+
+    /**
+     * Load the struction into the AI.
+     */
+    private void loadStructure()
+    {
+        final WorkOrderBuildDecoration workOrder = job.getWorkOrder();
+
+        if (workOrder == null)
+        {
+            return;
+        }
+
+        final BlockPos pos = workOrder.getBuildingLocation();
+        if (workOrder instanceof WorkOrderBuildBuilding && worker.getColony().getBuildingManager().getBuilding(pos) == null)
+        {
+            Log.getLogger().warn("AbstractBuilding does not exist - removing build request");
+            worker.getColony().getWorkManager().removeWorkOrder(workOrder);
+            return;
+        }
+
+        final int tempRotation = workOrder.getRotation(world);
+        final boolean removal = workOrder instanceof WorkOrderBuildRemoval;
+
+        loadStructure(workOrder.getStructureName(), tempRotation, pos, workOrder.isMirrored(), removal);
+        workOrder.setCleared(false);
+        workOrder.setRequested(false);
+
+        //We need to deal with materials
+        requestMaterials();
+    }
+
+    /**
+     * Iterates through all the required resources and stores them in the building.
+     * Suppressing Sonar Rule Squid:S135
+     * The rule thinks we should have less continue and breaks.
+     * But in this case the rule does not apply because code would become unreadable and uneffective without.
+     */
+    @SuppressWarnings(LOOPS_SHOULD_NOT_CONTAIN_MORE_THAN_A_SINGLE_BREAK_OR_CONTINUE_STATEMENT)
+    private void requestMaterials()
+    {
+        if (Configurations.gameplay.builderInfiniteResources || job.getWorkOrder().isRequested() || job.getWorkOrder() instanceof WorkOrderBuildRemoval)
+        {
+            return;
+        }
+
+        final AbstractBuildingStructureBuilder buildingWorker = getOwnBuilding(AbstractBuildingStructureBuilder.class);
+        buildingWorker.resetNeededResources();
+
+        while (job.getStructure().findNextBlock())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)